apps/e2e/ag-grid-storybook/src/app/ag-grid/ag-grid-stories.component.ts

```diff
@@ -1,79 +1,255 @@
-import { SkyAgGridService, SkyCellType } from '@skyux/ag-grid';
+import {
+  SkyAgGridRowDeleteConfirmArgs,
+  SkyAgGridService,
+  SkyCellType,
+} from '@skyux/ag-grid';
+import { SkyDockLocation, SkyDockService } from '@skyux/core';
+import { SkyThemeService, SkyThemeSettings } from '@skyux/theme';
 
-import { GridOptions } from 'ag-grid-community';
-import { BehaviorSubject, timer } from 'rxjs';
-import { first } from 'rxjs/operators';
+import {
+  Events,
+  GridOptions,
+  RowNode,
+  RowSelectedEvent,
+} from 'ag-grid-community';
+import { BehaviorSubject, Observable, Subscription, combineLatest } from 'rxjs';
+import { delay, filter, map } from 'rxjs/operators';
 
 import { columnDefinitions, data } from '../shared/baseball-players-data';
 
+import { ContextMenuComponent } from './context-menu.component';
+
+type DataSet = { id: string; data: any[] };
+
 @Component({
   selector: 'app-ag-grid-stories',
   templateUrl: './ag-grid-stories.component.html',
   styleUrls: ['./ag-grid-stories.component.scss'],
   encapsulation: ViewEncapsulation.None,
 })
-export class AgGridStoriesComponent implements OnInit {
-  @HostBinding('class.use-normal-dom-layout')
-  public get useNormalDomLayout(): boolean {
-    return this.domLayout === 'normal';
+export class AgGridStoriesComponent
+  implements OnInit, AfterViewInit, OnDestroy
+{
+  public dataSets: DataSet[] = [
+    {
+      id: 'back-to-top',
+      data: data.slice(40, 48),
+    },
+    {
+      id: 'row-delete',
+      data: data.slice(50, 53),
+    },
+    {
+      id: 'validation',
+      data: data.slice(60, 63),
+    },
+  ];
+  public gridOptions: { [_: string]: GridOptions } = {};
+  public isActive$ = new BehaviorSubject(true);
+  public addPreviewWrapper$ = new BehaviorSubject(false);
+  public ready = new BehaviorSubject(false);
+  public rowDeleteIds: string[] = [];
+  public skyTheme: SkyThemeSettings;
+
+  readonly #gridsReady = new Map<string, Observable<boolean>>();
+  readonly #agGridService: SkyAgGridService;
+  readonly #themeSvc: SkyThemeService;
+  readonly #changeDetectorRef: ChangeDetectorRef;
+  readonly #dockService: SkyDockService;
+  readonly #doc: Document;
+  readonly #ngUnsubscribe: Subscription;
+
+  constructor(
+    agGridService: SkyAgGridService,
+    themeSvc: SkyThemeService,
+    changeDetectorRef: ChangeDetectorRef,
+    dockService: SkyDockService,
+    @Inject(DOCUMENT) doc: Document
+  ) {
+    this.#agGridService = agGridService;
+    this.#themeSvc = themeSvc;
+    this.#changeDetectorRef = changeDetectorRef;
+    this.#dockService = dockService;
+    this.#doc = doc;
+    this.#ngUnsubscribe = new Subscription();
   }
 
-  @HostBinding('class.use-auto-height-dom-layout')
-  public get useAutoHeightDomLayout(): boolean {
-    return this.domLayout === 'autoHeight';
+  public ngOnInit(): void {
+    this.#dockService.setDockOptions({
+      location: SkyDockLocation.ElementBottom,
+      referenceEl: this.#doc.querySelector('#back-to-top'),
+    });
```

Comment:
Attach back-to-top to its respective grid rather than the whole screen as it would by default.